lib/data/repository/repository.dart

```diff
@@ -0,0 +1,3 @@
+abstract class Repository {
+  Future<void> registerUser(String password, String email);
```

Comment:
are you sure about Future<void> reutrn type? mby we need to return Future<bool> for case if user registered or not? or something else, void is too uninformative